### PR TITLE
#5858 Ray.java normalize direction vector in set() methods

### PR DESCRIPTION
**Pull request by fabiitch**
finnally ! 

nor() direction vector everytime.
Cause Math.sqrt avoid if vector is already normalized in Vector2#nor().
If you want to avoid the call of nor() , use the public field direction.

thx for reading.

### CODE REVIEW

File: gdx/src/com/badlogic/gdx/math/collision/Ray.java [hunk 1]
```diff
@@ -104,7 +104,7 @@ public Ray set (float x, float y, float z, float dx, float dy, float dz) {
 	 * @return This ray for chaining */
 	public Ray set (Ray ray) {
 		this.origin.set(ray.origin);
-		this.direction.set(ray.direction);
+		this.direction.set(ray.direction).nor();
```
Comment (obigu) [inline gdx/src/com/badlogic/gdx/math/collision/Ray.java]: It's minor but there's no need to normalize as the passed ray will have its direction already normalized

**Comment by obigu** [comment]
To make sure Ray instances have their direction always normalized (except if manipulating via the public field) I think we should also add normalize to `Ray mul (Matrix4 matrix)` in case the matrix scales.

File: gdx/src/com/badlogic/gdx/math/collision/Ray.java (hunk 1 above)
Comment (fabiitch) [reply gdx/src/com/badlogic/gdx/math/collision/Ray.java]: ye, i got this interogation of the utility in this case. 
I add the nor() for be similar with other method.

**Comment by fabiitch** [comment]
effectively ,
Ray mul (Matrix4 matrix) should included in this pr.

will ammend commit soon.

**Comment by noblemaster** [comment]
We should also add the change to the "CHANGES" text file! :)

**Comment by fabiitch** [comment]
Changelog edited !

File: CHANGES [hunk 2]
```diff
@@ -3,6 +3,7 @@
 - Update to LWJGL 3.2.3
 - API Change: Table#getRow now returns -1 when over the table but not over a row (used to return the last row).
 - API Change: Tree#addToTree and #removeFromTree now have an "int actorIndex" parameter.
+- API Change: Ray#set methods and Ray#mul(Matrix4) normalize direction vector.Use public field to set and avoid nor()
```
Comment (intrigus) [inline CHANGES]: Nitpick: Missing space between "vector.Use public field".

**Comment by noblemaster** [comment]
Looks fine to me now. Any comments, reservations or is this ready to be merged?

File: CHANGES (hunk 2 above)
Comment (noblemaster) [reply CHANGES]: It's now fixed.

**Comment by intrigus** [comment]
LGTM

**Pull request merged**